app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,8 @@
+<h1><%= @recipe.name %></h1>
+  <p> Complexity(1-10): <%= @recipe.complexity %> </p>
+  <p> <%= @recipe.genre %> </p>
+  <p> <% @recipe_ingredients.each do |ingredient| %> </p>
+    <p> <%= ingredient.name %> </p>
+    <% end %>
+
+    <p> Total Cost: <%= @recipe_ingredients.sum(:cost)%> </p>
```

Comment:
Two issues with this line: 
1. Using `@recipe_ingredients` as you implemented in the controller is actually using all ingredients, which is the same problem as I listed above. 
2. Doing .`sum(:cost)` is the right approach for this query, but we're doing it in the view, which is an MVC infraction. A good rule of thumb is to remember to move any AR to a model method and call that method where it needs to be, and then test that model method as well inside a model test file.

Reply:
Finally, when you have put this functionality in a model method, it should be tested. Another good thing to remember is that if you finish an IC without any new model methods, you should go back and double-check you're not making any accidental MVC infractions!